components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/utils/SCIMCommonUtils.java

```diff
@@ -66,39 +70,54 @@
      */
     private static ThreadLocal<Boolean> threadLocalIsManagedThroughSCIMEP = new ThreadLocal<>();
 
-    public static String getSCIMUserURL(String id) {
+    public static String getSCIMUserURL(String id) throws IdentitySCIMException {
+
         return StringUtils.isNotBlank(id) ? getSCIMUserURL() + SCIMCommonConstants.URL_SEPERATOR + id : null;
     }
 
-    public static String getSCIMGroupURL(String id) {
+    public static String getSCIMGroupURL(String id) throws IdentitySCIMException {
+
         return StringUtils.isNotBlank(id) ? getSCIMGroupURL() + SCIMCommonConstants.URL_SEPERATOR + id : null;
     }
 
-    public static String getSCIMServiceProviderConfigURL(String id){
+    public static String getSCIMServiceProviderConfigURL(String id) throws IdentitySCIMException {
+
         return getSCIMServiceProviderConfigURL() ;
     }
 
     /*Handling ThreadLocals*/
 
-    public static String getSCIMUserURL() {
+    public static String getSCIMUserURL() throws IdentitySCIMException {
+
         String scimURL = getSCIMURL();
         return scimURL + SCIMCommonConstants.USERS;
     }
 
-    public static String getSCIMGroupURL() {
+    public static String getSCIMGroupURL() throws IdentitySCIMException {
+
         String scimURL = getSCIMURL();
         return scimURL + SCIMCommonConstants.GROUPS;
     }
 
-    private static String getSCIMURL() {
+    private static String getSCIMURL() throws IdentitySCIMException {
+
         String scimURL;
-        String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
-        if (isNotASuperTenantFlow(tenantDomain)) {
-            scimURL = IdentityUtil.getServerURL(
-                    SCIMCommonConstants.TENANT_URL_SEPERATOR + tenantDomain + SCIMCommonConstants.SCIM2_ENDPOINT, true,
-                    true);
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+            try {
+                scimURL = ServiceURLBuilder.create().addPath(SCIMCommonConstants.SCIM2_ENDPOINT).build()
+                        .getAbsolutePublicURL();
+            } catch (URLBuilderException e) {
+                throw new IdentitySCIMException("Error occurred while building the SCIM2 endpoint with tenant " +
+                        "qualified URL.", e);
+            }
         } else {
-            scimURL = IdentityUtil.getServerURL(SCIMCommonConstants.SCIM2_ENDPOINT, true, true);
+            String tenantDomain = PrivilegedCarbonContext.getThreadLocalCarbonContext().getTenantDomain();
+            if (isNotASuperTenantFlow(tenantDomain)) {
+                scimURL = IdentityUtil.getServerURL(SCIMCommonConstants.TENANT_URL_SEPERATOR + tenantDomain +
+                        SCIMCommonConstants.SCIM2_ENDPOINT, true, true);
+            } else {
+                scimURL = IdentityUtil.getServerURL(SCIMCommonConstants.SCIM2_ENDPOINT, true, true);
+            }
```

Comment:
Do we need this part? Inside the ServiceURLBuilder also it's checking the `isTenantQualifiedUrlsEnabled()` config. So if there is no additional config file used we don't need to specifically check the `isTenantQualifiedUrlsEnabled()` again here. If so we can get rid of the old `IdentityUtil.getServerURL()` completely.

Reply:
If the configuration is disabled ServiceURLBuilder doesn't return the tenant domain

Reply:
So in the legacy way, we still had the tenant qualified complete url built ?

Reply:
yes